src/persist-client/src/read.rs

```diff
@@ -319,13 +318,12 @@ where
     /// `as_of` that would have been accepted.
     pub async fn listen(&self, as_of: Antichain<T>) -> Result<Listen<K, V, T, D>, Since<T>> {
         trace!("ReadHandle::listen as_of={:?}", as_of);
-        if PartialOrder::less_than(&as_of, &self.since) {
-            return Err(Since(self.since.clone()));
-        }
+        let mut machine = self.machine.clone();
+        let () = machine.listen(&as_of).await?;
```

Comment:
this change hangs the open_loop benchmark because it creates listeners as_of zero at startup and immediately awaits them before any writes have come in. any ideas how to proceed? everything I've come up with has been awful. https://github.com/MaterializeInc/materialize/blob/a0ec852470cfe30e5d18b41a4080ed7fcedcf1fd/src/persist-client/examples/open_loop.rs#L441-L443

Reply:
Maybe this is an early sign that the previous behaviour of `listen()` was more ergonomic? (See my other comment)